src/renderer/store/downloadModel/reducer.ts

```diff
@@ -18,10 +21,35 @@ const downloadModelReducer: Reducer<
       isDownloading: true,
       isDownloadComplete: false,
       downloadProgress: 0,
+      lastUpdated: new Date(),
+      previousDownloadProgress: 0,
+      timeRemaining: null,
     } as DownloadModel;
   }
 
   if (action.type === DOWNLOAD_PROGRESS_UPDATE) {
+    const newLastUpdated = new Date();
```

Comment:
Reducers should be deterministic pure functions so that they are replayable, this breaks that invariant - can instead pass in new date as an argument to the action

Reply:
Good point. I've altered the reducer to not be deterministic. However, I didn't want frontend or handler code to have to worry about passing in the date in the action dispatch so its defined inside the action as that is essentially a function. It is no longer defined in the reducer. Thank you! 😃

Reply:
Yep this is the way to go, nice job. Just on terminology it IS deterministic now not the other way round :)

Reply:
oh lol thank you